go.mod

```diff
@@ -16,14 +16,14 @@ require (
 	connectrpc.com/connect v1.17.0
 	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.16.0
 	github.com/Azure/azure-sdk-for-go/sdk/azidentity v1.8.0
-	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v3 v3.0.1
-	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v2 v2.4.0
+	github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v6 v6.1.0
```

Comment:
These are big version bumps, are there any notable changes in between the versions?

Are we bumping because the older versions are discontinued in favor of the newer?

Reply:
tl;dr the only breaking changes are renaming/removing types as the Azure API evolves and rebrands stuff. It seems the go sdk is mostly generated code and the version matches the Azure API version. If something that we use changed, we would not be able to compile anymore.

Changelog details:

### armredis breaking changes

#### v3

- Type of OperationStatus.Error has been changed from *ErrorDetailAutoGenerated to *ErrorDetail
- Type of OperationStatusResult.Error has been changed from *ErrorDetailAutoGenerated to *ErrorDetail
Struct ErrorDetailAutoGenerated has been removed

### armcompute breaking changes

#### v6

- Type of SecurityPostureReference.ExcludeExtensions has been changed from []*VirtualMachineExtension to []*string

#### v5

- Type of CommunityGalleryImageProperties.Identifier has been changed from *GalleryImageIdentifier to *CommunityGalleryImageIdentifier
- Type of GalleryTargetExtendedLocation.StorageAccountType has been changed from *StorageAccountType to *EdgeZoneStorageAccountType
- Type of RestorePointSourceVMDataDisk.DiskRestorePoint has been changed from *APIEntityReference to *DiskRestorePointAttributes
- Type of RestorePointSourceVMOSDisk.DiskRestorePoint has been changed from *APIEntityReference to *DiskRestorePointAttributes
- StorageAccountTypeStandardSSDLRS from enum StorageAccountType has been removed
- Field ID of struct VirtualMachineScaleSetIPConfiguration has been removed
- Field ID of struct VirtualMachineScaleSetNetworkConfiguration has been removed
- Field ID of struct VirtualMachineScaleSetUpdateIPConfiguration has been removed
- Field ID of struct VirtualMachineScaleSetUpdateNetworkConfiguration has been removed

#### v4

- Type of GalleryImageVersionStorageProfile.Source has been changed from *GalleryArtifactVersionSource to *GalleryArtifactVersionFullSource
- Type of SharingProfile.CommunityGalleryInfo has been changed from interface{} to *CommunityGalleryInfo
- Type of VirtualMachineExtensionUpdateProperties.ProtectedSettingsFromKeyVault has been changed from interface{} to *KeyVaultSecretReference
- Type of GalleryOSDiskImage.Source has been changed from *GalleryArtifactVersionSource to *GalleryDiskImageSource
- Type of GalleryDiskImage.Source has been changed from *GalleryArtifactVersionSource to *GalleryDiskImageSource
- Type of GalleryDataDiskImage.Source has been changed from *GalleryArtifactVersionSource to *GalleryDiskImageSource
- Type of VirtualMachineScaleSetExtensionProperties.ProtectedSettingsFromKeyVault has been changed from interface{} to *KeyVaultSecretReference
- Type of VirtualMachineExtensionProperties.ProtectedSettingsFromKeyVault has been changed from interface{} to *KeyVaultSecretReference
- Field URI of struct GalleryArtifactVersionSource has been removed

#### v3

- Function *CloudServicesClient.BeginCreateOrUpdate parameter(s) have been changed from (context.Context, string, string, *CloudServicesClientBeginCreateOrUpdateOptions) to (context.Context, string, string, CloudService, *CloudServicesClientBeginCreateOrUpdateOptions)
- Function *CloudServicesClient.BeginUpdate parameter(s) have been changed from (context.Context, string, string, *CloudServicesClientBeginUpdateOptions) to (context.Context, string, string, CloudServiceUpdate, *CloudServicesClientBeginUpdateOptions)
- Function *CloudServicesUpdateDomainClient.BeginWalkUpdateDomain parameter(s) have been changed from (context.Context, string, string, int32, *CloudServicesUpdateDomainClientBeginWalkUpdateDomainOptions) to (context.Context, string, string, int32, UpdateDomain, *CloudServicesUpdateDomainClientBeginWalkUpdateDomainOptions)
- Type of CloudServiceExtensionProperties.Settings has been changed from *string to interface{}
- Type of CloudServiceExtensionProperties.ProtectedSettings has been changed from *string to interface{}
- Field Parameters of struct CloudServicesClientBeginUpdateOptions has been removed
- Field Parameters of struct CloudServicesClientBeginCreateOrUpdateOptions has been removed
- Field Parameters of struct CloudServicesUpdateDomainClientBeginWalkUpdateDomainOptions has been removed

### armcontainerservice breaking changes

#### v6
- Function *ManagedClustersClient.GetOSOptions has been removed
- Struct OSOptionProfile has been removed
- Struct OSOptionProperty has been removed
- Struct OSOptionPropertyList has been removed

#### v5
- Field NodeSelector of struct IstioEgressGateway has been removed

#### v4
- Field DockerBridgeCidr of struct NetworkProfile has been removed

#### v3
- Const ManagedClusterSKUNameBasic from type alias ManagedClusterSKUName has been removed
- Const ManagedClusterSKUTierPaid from type alias ManagedClusterSKUTier has been removed

Reply:
Many thanks for the details, folks.